### PR TITLE
align eip-1898: support block number or hash on state-related methods

### PR DESCRIPTION
**Pull request by peekpi**
merge https://github.com/ethereum/go-ethereum/pull/19491 to harmony.

### CODE REVIEW

File: rpc/blockchain.go [hunk 1]
```diff
@@ -803,23 +803,9 @@ func (s *PublicBlockchainService) GetProof(
...
-		err = ErrRequestedBlockTooHigh
-		return
-	}
-
-	// Fetch Header
-	header, err := s.hmy.HeaderByNumber(ctx, blockNum)
-	if header == nil && err != nil {
-		return
-	}
-	state, err := s.hmy.BeaconChain.StateAt(header.Root())
```
Comment (peekpi) [inline rpc/blockchain.go:820]: @rlan35 do you know why here use BeaconChain but BlockChain? what's the difference between them?

File: rosetta/services/account.go [hunk 2]
```diff
@@ -69,7 +69,7 @@ func (s *AccountAPI) AccountBalance(
 			return nil, rosettaError
 		}
 	} else {
-		balance, err = s.hmy.GetBalance(ctx, addr, blockNum)
+		balance, err = s.hmy.GetBalance(ctx, addr, rpc.BlockNumberOrHashWithNumber(blockNum))
```
Comment (peekpi) [inline rosetta/services/account.go:72]: @LuttyYang please help to check the change of rosetta

**Comment by sophoah** [comment]
@peekpi can you attach rpc call result test in the PR showing we still support the old way along with the new ones

**Comment by peekpi** [comment]
> @peekpi can you attach rpc call result test in the PR showing we still support the old way along with the new ones

travis_rpc_test has some tests for the old RPC.

**Comment by sophoah** [comment]
@peekpi Tested on mainnet archival node

## by block number
```
curl -s -L -X POST $URL --header 'Content-Type: application/json' --data-raw "{\"jsonrpc\":\"2.\":\"eth_call\", \"params\":[{\"to\": \"0xcf664087a5bb0237a0bad6742852ec6c8d69a27a\",\"data\": \"0x70a0823100000000000000000000000083652856fc0431f6f735fdb271c468d0dd41fd4c\"}, \"0x19BE6C2\"],\"id\":1}"
{"jsonrpc":"2.0","id":1,"result":"0x0000000000000000000000000000000000000000000000011bada83bc6ed2c45"}
```
## by blockhash inside an object
```
curl -s -L -X POST $URL --header 'Content-Type: application/json' --data-raw "{\"jsonrpc\":\"2.0\",\"method\":\"eth_call\",\"params\":[{\"to\": \"0xcf664087a5bb0237a0bad6742852ec6c8d69a27a\",\"data\": \"0x70a0823100000000000000000000000083652856fc0431f6f735fdb271c468d0dd41fd4c\"}, {\"blockHash\":\"0xc03e94bf147b041afe181aef60fcdffec30d8fa794359f5f8d89c1bd9ea9e65e\"}],\"id\":1}"
{"jsonrpc":"2.0","id":1,"result":"0x0000000000000000000000000000000000000000000000011bada83bc6ed2c45"}
```

**Review by rlan35 — APPROVED** [review]

**Pull request merged**